src/test/unit/test/controllers/UserSsoController.ts

```diff
@@ -0,0 +1,114 @@
+
+describe('User remove SSO controller', () => {
+  mockRootController();
+  let req: any;
+  const res = mockResponse();
+  const controller = new UserSsoController();
+
+  beforeEach(() => {
+    req = mockRequest();
+    req.scope.cradle.api = mockApi;
+  });
+
+  test('Should render the user remove sso page', async () => {
+    const userData = {
+      id: 1,
+      forename: 'John',
+      surname: 'Smith',
+      email: 'john.smith@test.local',
+      active: true,
+      roles: ['IDAM_SUPER_USER'],
+    };
+
+    req.body = { _userId: userData.id };
+    when(mockApi.getUserById).calledWith(userData.id).mockReturnValue(Promise.resolve(userData));
+
+    await controller.post(req, res);
+    expect(res.render).toBeCalledWith('sso-user', { content: { user: userData } });
+  });
+
+  test('Should redirect to the user manage page after removing sso for user', async () => {
+    const userData = {
+      id: 1,
+      forename: 'John',
+      surname: 'Smith',
+      email: 'john.smith@test.local',
+      active: true,
+      roles: ['IDAM_SUPER_USER'],
+    };
+
+    req.body = { _userId: userData.id, _action: 'confirm-remove-sso', confirmSso: 'true' };
+    when(mockApi.getUserById).calledWith(userData.id).mockReturnValue(Promise.resolve(userData));
+    when(mockApi.deleteSsoById).calledWith(userData.id).mockReturnValue(Promise.resolve());
+
+    await controller.post(req, res);
+    expect(res.render).toBeCalledWith('sso-user-successful', { content: { user: userData } });
+  });
+
+  test('Should redirect to the user details page after cancelling removing sso for user', async () => {
+    const userData = {
+      id: 1,
+      forename: 'John',
+      surname: 'Smith',
+      email: 'john.smith@test.local',
+      active: true,
+      roles: ['IDAM_SUPER_USER'],
+    };
+
+    req.body = { _userId: userData.id, _action: 'confirm-remove-sso', confirmSso: 'false' };
+    when(mockApi.getUserById).calledWith(userData.id).mockReturnValue(Promise.resolve(userData));
+
+    await controller.post(req, res);
+    expect(res.redirect).toBeCalledWith(307, USER_DETAILS_URL);
+  });
+
+  test('Should render the remove sso page with validation errors after confirming', async () => {
+    const userData = {
+      id: 1,
+      forename: 'John',
+      surname: 'Smith',
+      email: 'john.smith@test.local',
+      active: true,
+      roles: ['IDAM_SUPER_USER'],
+    };
+
+    const error = { confirmRadio: { message: MISSING_OPTION_ERROR } };
+
+    when(mockApi.getUserById).calledWith(userData.id).mockReturnValue(Promise.resolve(userData));
+    req.body = { _userId: userData.id, _action: 'confirm-remove-sso' };
+
+    await controller.post(req, res);
+    expect(mockApi.getUserById).toBeCalledWith(userData.id);
+    expect(res.render).toBeCalledWith('sso-user', { content: { user: userData }, error });
+  });
+
+  test('Should render the remove sso page after there was an API issue', async () => {
```

Comment:
The description of test should be 'Should render the remove sso page **with error**......'

Reply:
This description is consistent with a similar test on other functionality like delete